cmd/mt-kafka-mdm-report-out-of-order/inputooofinder.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/grafana/metrictank/idx/cassandra"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/schema/msg"
+	log "github.com/sirupsen/logrus"
+)
+
+type Track struct {
+	Name string
+	Tags []string
+
+	LatestTimestamp int64
+}
+
+type Tracker map[schema.MKey]Track
+
+// find out of order metrics
+type inputOOOFinder struct {
+	prefix string
+	substr string
+
+	tracker                  Tracker
+	reorderBufferGracePeriod int64
```

Comment:
I think there was a misunderstanding: This should not be a single value for all series. This should  be dependent on the incoming series interval. For example, given reorder buffer of size ;  if the series has interval 30s, then data 150 seconds or more older than most recently seen would be discarded (counted as OOO). However, data with interval 1s would only have a window of 4s (5s would be considered too old).

Reply:
Ah, this makes much more sense. I thought this was a property of the buffer.